Dockerfile

```diff
@@ -1,65 +1,102 @@
     apt-get update -qq && \
-    apt-get install --no-install-recommends -y mercurial python-psycopg2 sudo python-virtualenv \
-    libyaml-dev libffi-dev libssl-dev \
-    curl git python-pip python-gnuplot python-psutil && \
+    apt-get install --no-install-recommends -y \
+              curl \
+              git \
+              libffi-dev \
+              libssl-dev \
+              libyaml-dev \
+              mercurial \
+              python-gnuplot \
+              python-pip \
+              python-psutil \
+              python-psycopg2 \
+              python-virtualenv \
+              gcc-4.6 \
+              python-dev \
+              sudo \
+    && \
     pip install --upgrade pip && \
     apt-get purge -y software-properties-common && \
     apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-RUN git clone --depth 1 --single-branch --branch rel_17.05_plus_k8s_fs_support_resource_limit_reqs https://github.com/phnmnl/galaxy.git
-WORKDIR galaxy
+
+# Clone galaxy into /galaxy directory
+RUN git clone --depth 1 --single-branch --branch release_17.09_plus_isa_k8s_resource_limts https://github.com/phnmnl/galaxy.git
+WORKDIR /galaxy
+
 RUN echo "pykube==0.15.0" >> requirements.txt
-COPY config/galaxy.ini config/galaxy.ini
-COPY config/job_conf.xml config/job_conf.xml
-COPY config/tool_conf.xml config/tool_conf.xml
-COPY config/sanitize_whitelist.txt config/sanitize_whitelist.txt
-COPY config/job_resource_params_conf.xml config/job_resource_params_conf.xml
-COPY config/phenomenal_tools2container.yaml config/phenomenal_tools2container.yaml
 
-# Galaxy tours which guide users through the subsequent steps in an analysis
-COPY config/plugins/tours/metfrag+xcms-tour.yaml config/plugins/tours/metfrag+xcms-tour.yaml
+COPY config/datatypes_conf.xml \
+     config/galaxy.ini \
+     config/job_conf.xml \
+     config/job_resource_params_conf.xml \
+     config/phenomenal_tools2container.yaml\
+     config/sanitize_whitelist.txt \
+     config/tool_conf.xml \
+  config/
 
 COPY rules/k8s_destinations.py lib/galaxy/jobs/rules/k8s_destination.py
 COPY tools/phenomenal tools/phenomenal
 
 RUN virtualenv .config_script_venv
 RUN /bin/bash -c "source .config_script_venv/bin/activate && \
-                  pip install bioblend==0.9.0 && \
+                  pip install bioblend>=0.9.0 && \
                   deactivate"
 
+RUN ln -s /usr/bin/x86_64-linux-gnu-gcc-4.6 /usr/bin/x86_64-linux-gnu-gcc
+
+# Clone isatools-lite (Python 2 version)
+RUN git clone --depth 1 --single-branch --branch py2_isatools-lite https://github.com/ISA-tools/isa-api.git /isatools
+
 RUN virtualenv .venv
+# We provide --extra-index-url https://pypi.python.org/simple only until the Galaxy people
+# update their wheel server to include docutils==0.14, which Galaxy 17.09 requires.
 RUN /bin/bash -c "source .venv/bin/activate && \
```

Comment:
I would run the apt-get install part for gcc only here in this RUN, then continue with this pip installation on the virtualenv, and then on the same RUN at the end the purging commands for GCC.